src/libraries/System.Net.Quic/src/System/Net/Quic/QuicListener.cs

```diff
@@ -122,7 +134,8 @@ private unsafe QuicListener(QuicListenerOptions options)
 
         // Save the connection options before starting the listener
         _connectionOptionsCallback = options.ConnectionOptionsCallback;
-        _acceptQueue = Channel.CreateBounded<PendingConnection>(new BoundedChannelOptions(options.ListenBacklog) { SingleWriter = true });
+        _acceptQueue = Channel.CreateUnbounded<object>();
```

Comment:
any reason why don't leave it as Bounded channel?

Reply:
We impose the limits in a different fashion via `_pendingConnectionsCapacity` so no need to keep it bounded. And unbounded is more performant as it doesn't need to check the limits.

Reply:
I was mostly wondering if we could still relay on the channel counting instead of introducing new mechanisms. Bit it seems like you count immediately after the even and we also account the failures.
I think that is fine, we should be clear about it in https://learn.microsoft.com/en-us/dotnet/api/system.net.quic.quiclisteneroptions.listenbacklog?view=net-7.0#system-net-quic-quiclisteneroptions-listenbacklog as it differs from TCP e.g. may surprise callers.